### PR TITLE
Add PHP 8-only support (v2)

### PR DESCRIPTION
**Pull request by patinthehat**
This PR adds a new major version, v2.0.0.

Specifically, it:

- Removes support for all PHP 7.x versions.
- Requires PHP 8.0+.
- All syntax converted to PHP 8 where possible.
- Moves the service provider class to the `src` directory to match the structure of other packages.
- Implements `spatie/laravel-package-tools`.
- _Updates the changelog - release date needs to be updated, currently is "unreleased"._

### CODE REVIEW

**Comment by freekmurze** [comment]
@brendt could you review this one?

File: .php_cs.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...:true,"no_closing_tag":true,"no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_class_element_per_statement":{"elements":["property"]},"single_import_per_statement":true,"single_line_after_imports":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"visibility_required":true,"encoding":true,"full_opening_tag":true,"array_syntax":{"syntax":"short"},"ordered_imports":{"sortAlgorithm":"alpha"},"no_unused_imports":true,"not_operator_with_successor_space":true,"trailing_comma_in_multiline_array":true,"phpdoc_scalar":true,"unary_operator_spaces":true,"binary_operator_spaces":true,"blank_line_before_statement":{"statements":["break","continue","declare","return","throw","try"]},"phpdoc_single_line_var_spacing":true,"phpdoc_var_without_name":true},"hashes":{"src\/ViewModelsServiceProvider.php":2278103137,"src\/ViewModel.php":894200466,"src\/Console\/ViewModelMakeCommand.php":3721917699,"tests\/ViewModelTest.php":2153730530,"tests\/ViewModelMakeCommandTest.php":1388068081,"tests\/TestCase.php":4131197707,"tests\/DummyViewModel.php":3408407484}}
```
Comment (brendt) [inline .php_cs.cache:1]: Please add `.php_cs.cache` to your global gitignore and remove it:

```
git rm .php_cs.cache
git commit …
```

File: composer.json [hunk 2]
```diff
@@ -16,8 +16,9 @@
         }
     ],
     "require": {
-        "php": "^7.3|^8.0",
-        "illuminate/support": "^7.0|^8.0"
+        "php": "^8.0",
+        "illuminate/support": "^7.0|^8.0",
```
Comment (brendt) [inline composer.json:20]: Can't we drop `"illuminate/support": "^7.0` as well?

File: src/ViewModel.php [hunk 3]
```diff
@@ -39,7 +39,7 @@ public function toResponse($request): Response
         return new JsonResponse($this->items());
     }
 
-    public function view(string $view): ViewModel
+    public function view(string $view): self
```
Comment (brendt) [inline src/ViewModel.php:42]: Change `self` to `static`

**Review by brendt — CHANGES_REQUESTED** [review]

**Comment by brendt** [comment]
Thanks, I've retargetted this to the v2 branch and also gave some minor remarks.

File: .php_cs.cache (hunk 1 above)
Comment (patinthehat) [reply .php_cs.cache:1]: @brendt This file exists in the gitignore in [`spatie/package-skeleton-laravel`](https://github.com/spatie/package-skeleton-laravel/blob/master/.gitignore#L3) - would you prefer I merge that `.gitignore` into this package's `.gitiginore`?

**Review by brendt — APPROVED** [review]

**Comment by brendt** [comment]
Thanks! I'm not sure when I'll tag this new major yet, I'm first focussing on another package the coming days, but I'll come back to this one soon.

**Pull request merged**